turbopack/crates/turbo-tasks-memory/src/task.rs

```diff
@@ -1680,9 +1698,18 @@ impl Task {
         backend: &MemoryBackend,
         turbo_tasks: &dyn TurboTasksBackendApi<MemoryBackend>,
     ) {
-        let mut aggregation_context = TaskAggregationContext::new(turbo_tasks, backend);
         let mut state = self.full_state_mut();
         state.collectibles.emit(trait_type, collectible);
+        if let TaskStateType::InProgress(box InProgressState {
+            outdated_collectibles,
+            ..
+        }) = &mut state.state_type
+        {
+            if outdated_collectibles.remove_emit(trait_type, collectible) {
+                return;
+            }
+        }
```

Comment:
On recomputation it's possible that the same collectible is emitted again.

This adds an optimization, that if a collectible is an "outdated" collectible (a collectible that was emitted in the previous computation), we reuse that by removing it from "outdated" so we don't need to propagate it through the graph. So 0 propagations instead of 2 (add, remove).